src/components/SigningModals/TransactionStepper/useStepperLayout.js

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useRef, useState } from 'react'
+import useMeasure from 'react-use-measure'
+
+function useStepperLayout() {
+  const [outerBoundsRef, outerBounds] = useMeasure()
+  const innerBoundsRef = useRef()
+  const [innerBounds, setInnerBounds] = useState(null)
+  const [measuring, setMeasuring] = useState(true)
+
+  // First render must always be in "multiple" mode so that our measurement reference point is accurate
+  const [layout, setLayout] = useState('multiple')
+
+  useEffect(() => {
+    const outerMeasured = outerBounds.width && outerBounds.width > 0
+
+    if (!innerBounds) {
+      setInnerBounds(innerBoundsRef.current.offsetWidth)
```

Comment:
Is this state change synchronous (from what I understand, it's not—React groups all the state changes in a hook and applies them after?)? If not, maybe we should move it to its own `useEffect()`—the checks later may not make sense with a null `innerBounds`.

Reply:
This a great point, I _think_ placing this inside a `useLayoutEffect` will ensure execution order and remove the need to have the `measuring` flag at all